core/src/main/scala/filodb.core/downsample/ChunkDownsampler.scala

```diff
@@ -0,0 +1,272 @@
+                               startRow: Int,
+                               endRow: Int): Double = {
+
+    val vecPtr = chunkset.vectorPtr(colIds(0))
+    val colReader = part.chunkReader(colIds(0), vecPtr)
+    colReader.asDoubleReader.sum(vecPtr, startRow, endRow)
+  }
+}
+
+/**
+  * Downsamples by calculating count of values in one column
+  */
+case class CountDownsampler(override val colIds: Array[Int]) extends DoubleChunkDownsampler {
+  require(colIds.length == 1, s"Count downsample requires only one column. Got ${colIds.length}")
+  override val name: String = "dCount"
+  override def downsampleChunk(part: TimeSeriesPartition,
+                               chunkset: ChunkSetInfo,
+                               startRow: Int,
+                               endRow: Int): Double = {
+
+    val vecPtr = chunkset.vectorPtr(colIds(0))
+    val colReader = part.chunkReader(colIds(0), vecPtr)
+    colReader.asDoubleReader.count(vecPtr, startRow, endRow)
+  }
+}
+
+/**
+  * Downsamples by calculating min of values in one column
+  */
+case class MinDownsampler(override val colIds: Array[Int]) extends DoubleChunkDownsampler {
+  require(colIds.length == 1, s"Min downsample requires only one column. Got ${colIds.length}")
+  override val name: String = "dMin"
+  override def downsampleChunk(part: TimeSeriesPartition,
+                               chunkset: ChunkSetInfo,
+                               startRow: Int,
+                               endRow: Int): Double = {
+
+    val vecPtr = chunkset.vectorPtr(colIds(0))
+    val colReader = part.chunkReader(colIds(0), vecPtr)
+
+    var min = Double.MaxValue
+    var rowNum = startRow
+    val it = colReader.iterate(vecPtr, startRow).asDoubleIt
+    while (rowNum <= endRow) {
+      val nextVal = it.next
+      if (!JLDouble.isNaN(nextVal)) min = Math.min(min, nextVal)
+      rowNum += 1
+    }
+    min
+  }
+}
+
+/**
+  * Downsamples by calculating max of values in one column
+  */
+case class MaxDownsampler(override val colIds: Array[Int]) extends DoubleChunkDownsampler {
+  require(colIds.length == 1, s"Max downsample requires only one column. Got ${colIds.length}")
+  override val name: String = "dMax"
+  override def downsampleChunk(part: TimeSeriesPartition,
+                               chunkset: ChunkSetInfo,
+                               startRow: Int,
+                               endRow: Int): Double = {
+
+    val vecPtr = chunkset.vectorPtr(colIds(0))
+    val colReader = part.chunkReader(colIds(0), vecPtr)
+
+    var max = Double.MinValue
+    var rowNum = startRow
+    val it = colReader.iterate(vecPtr, startRow).asDoubleIt
+    while (rowNum <= endRow) {
+      val nextVal = it.next
+      if (!JLDouble.isNaN(nextVal)) max = Math.max(max, nextVal)
+      rowNum += 1
+    }
+    max
+  }
+}
+
+/**
+  * Downsamples by calculating average from average and count columns
+  */
+case class AvgAcDownsampler(override val colIds: Array[Int]) extends DoubleChunkDownsampler {
```

Comment:
What is the use case here?  I don't quite understand it.  Maybe add docs for the use case.